Actions/DetermineDeploymentEnvironments/DetermineDeploymentEnvironments.ps1

```diff
@@ -20,7 +20,7 @@ function IsGitHubPagesAvailable() {
 }
 
 function GetGitHubEnvironments() {
-    $headers = GetHeader -token $env:GITHUB_TOKEN
+    $headers = GetHeaders -token $env:GITHUB_TOKEN
```

Comment:
Small suggestion... Seems like we have a lot of places where we call GetHeaders and then call the GitHub API after. Maybe we can move it to a function that does both? Something like: 
`Invoke-GhApi -Uri "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/environments" -Token $env:GITHUB_TOKEN`

Reply:
Most places, we get the headers early because we reuse them, not sure it is worth the change to have a specific function for that.  Another change could be:
```
        $artifacts = (InvokeWebRequest -Headers (GetHeaders -token $token) -Uri $artifactsURI).Content | ConvertFrom-Json
```

Which would be fine the places where we use the headers once I guess.